Server/Temp.Domain/EventMessaging/DomainEventBus.cs

```diff
@@ -0,0 +1,84 @@
+﻿using ModularHouse.Server.Temp.Domain.EventMessaging.Contracts;
+using ModularHouse.Server.Temp.Domain.Exceptions;
+
+namespace ModularHouse.Server.Temp.Domain.EventMessaging;
+
+public class DomainEventBus : IDomainEventBus
+{
+    private record SubscribedEventHandler(Type EventType, Action<IDomainEvent> Callback)
+    {
+        public Guid SubscriptionId { get; } = Guid.NewGuid();
+    }
+
+    private readonly IList<SubscribedEventHandler> _subscribedHandlers = new List<SubscribedEventHandler>();
+    
+    public Task PublishAsync<TEvent>(TEvent domainEvent) where TEvent : IDomainEvent
+    {
+        var handlersToNotify = _subscribedHandlers
+            .Where(eventHandler => eventHandler.EventType == typeof(TEvent))
+            .ToList();
+
+        foreach (var eventHandler in handlersToNotify)
+            eventHandler.Callback(domainEvent);
+
+        return Task.CompletedTask;
+    }
+
+    public Task<Guid> SubscribeAsync<TEvent>(Action<TEvent> callback) where TEvent : IDomainEvent
+    {
+        var eventHandler = new SubscribedEventHandler(typeof(TEvent), AbstractedCallback);
+        _subscribedHandlers.Add(eventHandler);
+
+        return Task.FromResult(eventHandler.SubscriptionId);
+        
+        void AbstractedCallback(IDomainEvent evt) => callback((TEvent)evt);
+    }
+
+    public Task UnsubscribeAsync(Guid subscriptionId)
+    {
+        var eventHandler = _subscribedHandlers
+            .FirstOrDefault(handler => handler.SubscriptionId == subscriptionId);
```

Comment:
I think about, if we have 2+ handlers subscribed to the one topic.. Maybe remove them all ?

Reply:
Discussed in a call.